pennylane/ops/qubit.py

```diff
@@ -508,6 +546,73 @@ class CRot(Operation):
     grad_method = "A"
 
 
+class U2(Operation):
+    r"""U2(phi, lambda, wires)
+    U2 gate.
+
+    .. math::
+
+        U_2(\phi, \lambda) = \frac{1}{\sqrt{2}}\begin{bmatrix} 1 & -\exp(i \lambda)
+        \\ \exp(i \phi) & \exp(i (\phi + \lambda)) \end{bmatrix}
+
+    The :math:`U_2` gate is related to the single-qubit rotation :class:`Rot` and the
+    :class:`PhaseShift` gates via the following relation:
+
+    .. math::
+
+        U_2(\phi, \lambda) = R_\phi(\phi+\lambda) R(\lambda,\pi/2,-\lambda)
+
+    Args:
+        phi (float): azimuthal angle :math:`\phi`
+        lambda (float): quantum phase :math:`\lambda`
+        wires (Sequence[int] or int): the subsystem the gate acts on
+    """
+    num_params = 2
+    num_wires = 1
+    par_domain = "R"
+    grad_method = "A"
+
+    @staticmethod
+    def decomposition(phi, lam, wires=None):
+        Rot(lam, np.pi/2, -lam, wires=wires)
+        PhaseShift(lam, wires=wires)
+        PhaseShift(phi, wires=wires)
```

Comment:
Why not add `lam` and `phi` together?

Reply:
Yep, exactly. We would need to extend the variable instance to encapsulate addition of variables